Client/weather-app/src/components/Weather.vue

```diff
@@ -0,0 +1,471 @@
+            </div>
+        </div>
+            <p class="weather-history-label">Weather History</p>
+            <div class="weatherr-history-container">
+                    <div v-for="(weather, index) in weatherHistory" :key="weather.date" class="flex">
+                        <div v-for="(w) in weather" :key="w.date">
+                            <div class="weather">
+                                <div class="current">
+                                <div class="info">
+                                    <div>&nbsp;</div>
+                                    <div class="city">
+                                    <small>
+                                        <small>CITY:</small>
+                                    </small>
+                                    {{ w.city }}
+                                    </div>
+                                    <div class="temp">
+                                    <small>
+                                        <small>TEMP:</small>
+                                    </small>
+                                    {{w.temperature}}&deg;
+                                    <small>C</small>
+                                    </div>
+                                    <div class="wind">
+                                    <small>
+                                        <small>WIND:</small>
+                                    </small>
+                                    {{w.windSpeed}} km/h
+                                    </div>
+                                    <div class="humidity">
+                                    <small>
+                                        <small>HUMIDITY:</small>
+                                    </small>
+                                    {{w.humidity}}%
+                                    </div>
+                                    <div>&nbsp;</div>
+                                </div>
+                                <div class="icon">
+                                    <!-- <div class="sunny"></div> -->
+                                     <div :class=w.weatherDescription></div>
+                                </div>
+                                <div class="future">
+                                <div class="day" :class="{ 'current-weather': index === 0 }">
+                                    <h4 class="date">{{w.date}}</h4>
+                                </div>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+            </div>
+            </div>     
+</div>
+ 
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+    name: 'Weather',
+    
+    data () {
+        return {
+            weatherForecast: [],
+            inputValue: 'Bonn',
+            weatherHistory: [],
+            showError: false,
+            overcastCloudy: 'snowy'
+        }
+    },
+
+    computed: {
+        axiosParams() {
```

Comment:
`axiosParams` is not a good name. You should name things for what they represent, not how they are used. You're using this data for an axiom request but it represents the search parameters or search filters. A name `searchParameters` or `searchFilter` or `searchQuery` would be more appropriate.